core/src/util/usUtils.cpp

```diff
@@ -388,6 +388,37 @@ std::string GetLastErrorStr()
 #endif
 }
 
+//-------------------------------------------------------------------
+// Android Compatibility functions
+//-------------------------------------------------------------------
+
+namespace {
+
+template <typename T>
+std::string internal_to_string(T val)
```

Comment:
I think you can simplify this and remove the overloaded ```ToString()``` declarations by making ```ToString``` a template function. Doing this would also allow you to use the template ```ToString``` function in the test code.